deCODE/deCODE.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+# Import library for analysis-runner
+import click
+import hail as hl
+from cpg_utils.hail_batch import output_path
+
+# Import library for deCODE job
+import pandas as pd
+from pandas import Series,DataFrame
+import numpy as np
+import math
+import collections
+
+@click.command()
+@click.option('--dataset', help="data to query")
+@click.option('--output', help="output name")
+@click.option('--rerun', is_flag=True, help='Whether to overwrite cached files', default=False)
+def query(dataset, output, rerun):
+    p_out = output_path(output)
+    log_out = output_path(output + ".log")
+
+    hl.init(default_reference='GRCh38')
+    hl.context.warning("inited")
+    hl.copy_log(log_out)
+
+    if rerun or not hl.hadoop_exists(p_out):
+        # Section 1 - Read dataset
+        mt = hl.read_matrix_table(dataset)
+        mt = hl.experimental.densify(mt)
+        mt = hl.variant_qc(mt)        
+
+        # Section 2 - Sample-level QC
+        mt2 = hl.sample_qc(mt)
+       
+        # imputed sex equals to XX or XY
+        # Call rate >= 0.99
+        # mean coverage >= 20X
+        filter_conditions = ((mt2.meta.sex_karyotype == "XX") | (mt2.meta.sex_karyotype == "XY")) & \
+                            (mt2['sample_qc'].call_rate >= 0.99) & \
+                            (mt2['sample_qc'].dp_stats.mean >= 20)         
+        mt3 = mt2.filter_cols(filter_conditions, keep=True)
```

Comment:
Rather than creating a new matrix table for every step, would it be possible to essentially filter-in-place (i.e. `mt = hl.sample_qc(mt)` etc)? Whenever a mt table only needs to be used once (as input to the next filtering step), avoiding duplicating data into a new table can help with efficiency.

Reply:
Thanks for the suggestion! Done!